docs/migration.md

```diff
@@ -0,0 +1,127 @@
+
+If a plugin is not supported, it will be replaced by the Markdown panel in Perses with a note saying:
+`this panel is not supported`.
+
+## How it works
+
+As you might now, the Perses dashboard specification is written combining Golang and Cuelang.
+We rely on Cuelang when we need to describe a plugin (in the data-model). It allows us to have a dynamic and an
+extensible specification.
+
+The migration process is done in two parts:
+
+1. Step one: Import the Grafana Dashboard into a Golang structure and then migrate it to the Perses Golang structure.
+2. Step two: for each variables, panels and queries in the Grafana dashboard, we are executing a Cuelang script coming
+   from the plugin itself, if, of course, the plugin is supported. This script will generate the piece of the Perses
+   data-model for the corresponding plugin.
+
+If you need more information about how to write a migration script in Golang for a plugin, please refer to
+the [associated documentation](./plugins/cue.md#migration-from-grafana).
+
+## Prerequisites
+
+As we are always supporting the last version of Grafana, we advise to have your Grafana dashboard up to date if the
+latest version of Grafana.
+
+When we started to develop Perses, the latest version of Grafana was 9.0.0. And since we always have a backward
+compatible mind when developing the migration script. So there is a high chance you can migrate your dashboard from an
+older version of Grafana. But we can't guarantee it.
+
+## How to migrate
+
+1. First, you need a running instance of Perses. If you don't have one, please refer to
+   the [installation documentation](./installation/in-a-container.md).
+2. Then, you can either use the CLI or the UI to migrate your dashboard.
+
+### Using the UI
+
+- On the home page, click on the right of the button `add dashboard` and select `import dashboard`.
+- Then, you can either paste the JSON of your Grafana dashboard or upload the JSON file.
+- Click on the `import` button.
+- If the migration is successful, then you will have in return the Perses dashboard as a JSON.
+- You can save the JSON or continue on the same page to save this migrated dashboard into the project you would like.
+  You need to select the project and click on the `import` button.
+
+### Using the CLI
+
+- Install the CLI by following the [installation documentation](./cli.md).
+- Login the CLI to the Perses instance
+
+```bash
+percli login http://localhost:8080
+```
+
+- Then, you can use the `migrate` command to migrate your dashboard. We recommend using the `--online` flag to be sure
+  that the migration is done with the latest version of the plugins.
+
+```bash
+percli migrate -f grafana-dashboard.json --online -o json > perses-dashboard.json
+```
+
+- As a tip, you may want to open the file and remove any reference to the previous datasource used in Grafana. This
+  will allow the dashboard to use the default datasource. Do that only if you want to use the default datasource.
```

Comment:
@Nexucis Does it make sense to introduce a flag `--use-default-datasource=true` and if this flag is set then we don't add the datasource config at all in the dashboard manifest?

Reply:
That looks difficult to me to achieve that. The usage of the datasource is handled by Cuelang and by the cuelang migration script (so basically on every plugin).
Handling the Cuelang script was already something quite hard and it's executed more or less as a close box. And I'll  like to keep it like that.